lib/vfscore/vfs.h

```diff
@@ -69,89 +78,814 @@ extern int vfs_debug;
  * per task data
  */
 struct task {
-	char 	    t_cwd[PATH_MAX];	/* current working directory */
-	struct vfscore_file *t_cwdfp;		/* directory for cwd */
+	/* current working directory */
+	char t_cwd[PATH_MAX];
+	/* vfscore_file structure associated with current working directory */
+	struct vfscore_file *t_cwdfp;
 };
 
-int	 sys_open(char *path, int flags, mode_t mode, struct vfscore_file **fp);
-int	 sys_read(struct vfscore_file *fp, const struct iovec *iov, size_t niov,
+/**
+ * Opens the file specified by path.
+ *
+ * @param path
+ *	Path to the file to be opened
+ * @param flags
+ *	Flags that specifies how the file should be opened. It must contain one
+ *	of the following `access` modes:
+ *		* O_RDONLY  - uses the file only for reading purposes.
+ *		* O_WRONLY  - uses the file only for writing purposes.
+ *		* O_RDWR  - uses the file for both reading and writing.
+ *
+ *	Besides that, other flags can be specified as well:
+ *		* O_APPEND - positions the cursor at the end of the file before
+ *			the write operations.
+ *		* O_EXCL - if used with O_CREAT, it makes the sys_open() call
+ *			fail if the file already exists.
+ *		* O_NOFOLLOW - makes the sys_open() call fail if the final
+ *			component of the path refers to a symbolik link.
+ *		* O_TRUNC  - truncates a regular file to zero bytes if the file
+ *			was successfully opened in O_RDWR or O_WRONLY mode.
+ * @param mode
+ *	Mode bits to be applied when the new file is created
+ *	(these are ignored unless O_CREAT is provided in flags)
+ * @param[out] fp
+ *	Pointer to the task structure containing information about the opened
+ *	file on success, NULL on failure.
+ * @return
+ *	- (0):  Successfully initialized
```

Comment:
```suggestion
 *	- (0):  Successfully [...]
```
Change this `initialized` to whatever the function does. Do this for all the functions please.